### PR TITLE
Unwatch for onchaind

### PR DESCRIPTION
**Pull request by rustyrussell**
@cdecker reported we were using a heap of mem and falling behind in blocks which onchaind was spamming the logs with "Notified about tx %s output %u spend, but we don't care" messages.

Currently, we follow all txs from the funding tx onwards, in case onchaind is interested in them.  We have a FIXME to do something when onchaind says it's not interested, so this actually fixes that.

### CODE REVIEW

File: lightningd/watch.c [hunk 1]
```diff
@@ -112,6 +112,24 @@ struct txwatch *watch_txid_(const tal_t *ctx,
...
+
+	/* We could have more than one peer watching same txid, though we
+	 * don't for onchaind. */
+	for (w = txwatch_hash_getfirst(&topo->txwatches, txid, &i);
+	     w;
+	     w = txwatch_hash_getnext(&topo->txwatches, txid, &i)) {
+		if (w->peer == peer)
+			break;
+	}
+	return w;
```
Comment (cdecker) [inline lightningd/watch.c]: This always returns a `struct txwatch` even if it doesn't match the criteria, i.e., if the last one checked doesn't match the `peer` it still gets returned. That is not what [this check](https://github.com/ElementsProject/lightning/pull/833/commits/42073d4874733ea8aee6ebda783d4a0c5d7fe171#diff-b34700033c5cd47ee522323329a7a789R1207) assumes. Please return `NULL` if the watch cannot be found.

File: tests/test_lightningd.py [hunk 2]
```diff
@@ -2427,6 +2427,13 @@ def test_withdraw(self):
         c.execute('SELECT COUNT(*) FROM outputs WHERE status=0')
         assert(c.fetchone()[0] == 6)
 
+        # Test withdrawl to self.
```
Comment (cdecker) [inline tests/test_lightningd.py]: Nit: withdrawal

**Comment by cdecker** [comment]
I can confirm that this solves the `txwatch` fan-out problem we had on sleepyark :wink:

File: lightningd/watch.c (hunk 1 above)
Comment (rustyrussell) [reply lightningd/watch.c]: No, loop terminates on two conditions: w being NULL, or w->peer == peer.  The w != NULL check is buried in the loop conditional...

File: tests/test_lightningd.py (hunk 2 above)
Comment (rustyrussell) [reply tests/test_lightningd.py]: Oops, I can't even blame US vs British english.  Will fix...

**Pull request merged**